### PR TITLE
Adds a large clique size heuristic function

### PR DESCRIPTION
**Pull request by jfinkels**
This commit creates a new package, `networkx.algorithms.heuristic`, and
a new module within that package containing a function for finding the
size of a large clique in a graph.

This is the algorithm suggested in issue #773. I'm not certain if it is accurate, so please double check it.

There's no unit test because I'm not sure how to test it.

### CODE REVIEW

File: networkx/algorithms/heuristic/clique.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    :mod:`networkx.algorithms.clique`
+        Functions for finding the exact maximum clique in a graph.
+
+    """
+
+    def _clique_heuristic(G, U, size, best_size):
+        if not U:
+            return max(best_size, size)
+        u = max((v for v, d in G.degree(U)), key=itemgetter(1))
```
Comment (jtorrents) [inline networkx/algorithms/heuristic/clique.py]: I think this is more clear if written as:
```python
u = max(U, key=G.degree)
```

Comment (jfinkels) [inline networkx/algorithms/heuristic/clique.py]: Yes, I'll change that.

File: networkx/algorithms/heuristic/clique.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        U.remove(U)
```
Comment (jfinkels) [inline networkx/algorithms/heuristic/clique.py]: I think this was supposed to be `U.remove(u)` :).

Comment (hagberg) [reply networkx/algorithms/heuristic/clique.py]: ...still pending

**Comment by hagberg** [comment]
Needs tests.

**Pull request closed without merging**